tests/test_model_specs.py

```diff
@@ -57,59 +121,57 @@ def test_model_specs_are_valid(self):
         """MODEL_SPEC: test each spec meets the expected pattern."""
 
         required_keys = {'model_id', 'model_title', 'userguide',
-                         'aliases', 'args', 'ui_spec', 'outputs'}
-        optional_spatial_key = 'args_with_spatial_overlap'
+                         'aliases', 'inputs', 'ui_spec', 'outputs'}
         for model_id, pyname in model_id_to_pyname.items():
             model = importlib.import_module(pyname)
 
             # Validate top-level keys are correct
             with self.subTest(pyname):
                 self.assertTrue(
-                    required_keys.issubset(model.MODEL_SPEC),
+                    required_keys.issubset(set(dir(model.MODEL_SPEC))),
                     ("Required key(s) missing from MODEL_SPEC: "
-                     f"{set(required_keys).difference(model.MODEL_SPEC)}"))
-                extra_keys = set(model.MODEL_SPEC).difference(required_keys)
-                if (extra_keys):
-                    self.assertEqual(extra_keys, set([optional_spatial_key]))
+                     f"{set(required_keys).difference(set(dir(model.MODEL_SPEC)))}"))
+
+                if model.MODEL_SPEC.args_with_spatial_overlap:
```

Comment:
This whole module could be obsoleted by adding good type checking to the model spec classes, with or without pydantic